### DISCUSSION TITLE
Nextflow cannot find Java or it's a wrong version but it seems like I have a higher version

### DISCUSSION THREAD

**Discussion by das2000sidd** [nextflow-io/nextflow#3773] [Q&A]
I am trying to get nextflow installed but I am running into a Java version error and it seems like I have a higher version of Java installed. Was wondering if somebody can help me with installing nextflow.
Here is the output to java version check using `java --version `:

java 18.0.2.1 2022-08-18
Java(TM) SE Runtime Environment (build 18.0.2.1+1-1)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.2.1+1-1, mixed mode, sharing)

Here is the output to checking if nextflow works using `./nextflow run tutorial.nf`

Error: cannot find Java or it's a wrong version -- please make sure that Java 7 or higher it's installed
Note: Nextflow is trying to use the Java VM defined by the following environment variables:
 JAVA_CMD: /Library/Java/JavaVirtualMachines/jdk-18.0.2.1.jdk/Contents/Home/bin/java
 JAVA_HOME:

I am using a OS X operating system.

**Comment by pditommaso** [comment]
The suggested way to install Java is using [SdkMan](https://sdkman.io/)

**Comment by colinbrislawn** [comment]
I ran into this error when using conda envs and a too _new_ version of Java.

```sh
(qiime2-amplicon-2024.10) (base) >>> java --version
openjdk 22.0.1-internal 2024-04-16
OpenJDK Runtime Environment (build 22.0.1-internal-adhoc.conda.src)
OpenJDK 64-Bit Server VM (build 22.0.1-internal-adhoc.conda.src, mixed mode, sharing)

(qiime2-amplicon-2024.10) (base) >>> nextflow
Error: cannot find Java or it's a wrong version -- please make sure that Java 7 or higher it's installed
Note: Nextflow is trying to use the Java VM defined by the following environment variables:
 JAVA_CMD: /root/miniforge3/envs/qiime2-amplicon-2024.10/lib/jvm/bin/java
 JAVA_HOME: /root/miniforge3/envs/qiime2-amplicon-2024.10/lib/jvm
```

"Java 7 or higher " but not 22, that's too high!

Here's what works, and it's using jdk 11

```sh
(base) >>> java --version
openjdk 11.0.1 2018-10-16 LTS
OpenJDK Runtime Environment Zulu11.2+3 (build 11.0.1+13-LTS)
OpenJDK 64-Bit Server VM Zulu11.2+3 (build 11.0.1+13-LTS, mixed mode)

(base) >>> nextflow info
  Version: 24.10.3 build 5933
  Created: 16-12-2024 15:34 UTC 
  System: Linux 6.11.0-9-generic
  Runtime: Groovy 4.0.23 on OpenJDK 64-Bit Server VM 11.0.1+13-LTS
  Encoding: UTF-8 (UTF-8)
```
